src/expression/comparison/eq.ts

```diff
@@ -23,15 +23,10 @@ export class Equal extends Comparison {
     if (arguments.length !== 2) {
       throw new Error('comparison expression expects left and right operands')
     }
-    super('==', left, right)
+    super('==', OPERATOR, left, right)
   }
 
-  /**
-   * Evaluate in the given context.
-   * @param {Context} ctx
-   * @return {Result}
-   */
-  evaluate (ctx: Context): Result {
-    return this.left.evaluate(ctx) === this.right.evaluate(ctx)
+  comparison (left: Result, right: Result): boolean {
+    return left === right
   }
```

Comment:
why this change?
How this would work: ```[=, $Firstname, "Peter"]```?

Reply:
Oh I see, you've generalized the evaluation here: https://github.com/briza-insurance/illogical/blob/3b112351c786352cd5d4c4145af8bd51e294f8d9/src/expression/comparison/index.ts#L35-L37 ... make sense 👍

Reply:
Because we can reuse the comparison in the evaluate and simplify.

The parent implements evaluate that evaluates left and right passing them to comparison and returning its value. simplify will try to simplify left and right and if possible call comparison.